### PR TITLE
Redesigned workflow

### PR DESCRIPTION
**Pull request by teymurgahramanov**

### CODE REVIEW

File: actions/build-java-gradle/action.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+description: Building Java Projects with Gradle
+inputs:
+  report_path:
+    description: Relative path to directory containing JUnit reports
+    required: false
+outputs:
+  artifact-name:
+    description: "Artifact file name"
+    value: ${{ steps.build.outputs.artifact-name }}
+  artifact_path:
```
Comment (corest) [inline actions/build-java-gradle/action.yml:11]:
```suggestion
  artifact-path:
```

**Comment by corest** [comment]
@teymurgahramanov 

1. Please go over actions and update those where inputs names user `_` instead of `-`
2. What's the purpose of `proxies` folder?
3. Rename `java_gradle_app` to just `java_app`

**Comment by teymurgahramanov** [comment]
> @teymurgahramanov
> 
> 1. Please go over actions and update those where inputs names user `_` instead of `-`
> 2. What's the purpose of `proxies` folder?
> 3. Rename `java_gradle_app` to just `java_app`

What if we will use Maven later?  I think java_gradle_app is better in this case.

**Comment by corest** [comment]
@teymurgahramanov 

1. Check e.g. actions/build-java-gradle/action.yml

```
  report_path:
    description: Relative path to directory containing JUnit reports
    required: false
```

Go over other actions to confirm there are no underscores

2. Ack

3.

> What if we will use Maven later? I think java_gradle_app is better in this case.

Then we will just update it in new workflow version.
No need to keep this specific `gradle` thing in name

File: README.md [hunk 2]
```diff
@@ -1,283 +1,52 @@
...
+|       └── action.yml
+|       └── README.md
+└── proxies # Proxy workflows to be placed on project repository side
+|   └── proxy-workflow-name.yml # Must be the same as original workflow name
+```
+
+## Workflow naming
+
+${GITHUB_EVENT_NAME}_${LANGUAGE/STACK_MAIN_TYPE}_${LANGUAGE/STACK_SUB_TYPE}_${ENTITY}.yml
+Example: pull_request_java_gradle_app.yml
```
Comment (corest) [inline README.md:17]: update with proper name

File: .github/workflows/release-java-app.yml [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set Gradle release
+        id: set-java-release
+        uses: veegoteam/workflows/actions/set-java-release@DEVOPS-18
+        with:
+          current_version: ${{ needs.extract-java-metadata.outputs.version }}
```
Comment (corest) [inline .github/workflows/release-java-app.yml:71]:
```suggestion
          current-version: ${{ needs.extract-java-metadata.outputs.version }}
```

**Pull request merged**